util/clear.js

```diff
@@ -0,0 +1,36 @@
+'use strict'
+
+var setImmediate = require('./immediate')
+
+module.exports = function clear (db, location, keyRange, options, callback) {
+  if (options.limit === 0) return setImmediate(callback)
+
+  var transaction = db.db.transaction([location], 'readwrite')
+  var store = transaction.objectStore(location)
+  var count = 0
+
+  transaction.oncomplete = function () {
+    callback()
```

Comment:
Do we know for sure that `.oncomplete` and `.onabort` are called async?

Reply:
👍 Lemme check that

Reply:
The `complete` event is always async, because it can only happen after a transaction has started, which itself is async:

> the implementation must queue a task to start the transaction asynchronously
> https://www.w3.org/TR/IndexedDB-2/#transaction-lifetime-concept

I'm not sure about `abort`:

> A transaction can be **aborted at any time** before it is finished, even if the transaction isn’t currently active or **hasn’t yet started**. When a transaction is aborted the implementation must undo (roll back) any changes that were made to the database during that transaction. This includes both changes to the contents of object stores as well as additions and removals of object stores and indexes.
> https://www.w3.org/TR/IndexedDB-2/#transaction-lifetime-concept

But I think "can be aborted" here means "... by the user".

Reply:
I'm gonna assume that the spec authors thought about this scenario 😄